flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/DbSupport.java

```diff
@@ -144,6 +143,30 @@ public void restoreCurrentSchema() {
     protected abstract void doChangeCurrentSchemaTo(String schema) throws SQLException;
 
     /**
+     * Retrieves the current user.
+     *
+     * @return The current user for this connection.
+     */
+    public String getCurrentUserName() {
```

Comment:
User and Schema are different terms in most databases. This method should be the only correct way of determining the current user. `jdbcTemplate.getMetaData().getUser()` may lie (and it does for proxy connections). That's why this method should be considered for overriding in database-specific subclasses, and is overriden in OracleDbSupport class.

Absence of this method may lead (and actually did lead) to misuse of another method - `getCurrentSchemaName`, which was wrongly implemented (intentionally?) and works correctly in wrong context and incorrectly in right context.

This method is now used in doClean method instead of misused getCurrentSchemaName method.